src/org/opendatakit/briefcase/util/ExportToCsv.java

```diff
@@ -790,34 +793,25 @@ private boolean processInstance(File instanceDir) {
     EventBus.publish(new ExportProgressPercentageEvent((processedInstances * 100.0) / totalInstances, briefcaseLfd));
 
     // If we are encrypted, be sure the temporary directory
-    // that will hold the unencrypted files is created and empty.
+    // that will hold the unencrypted files is created.
     // If we aren't encrypted, the temporary directory
     // is the same as the instance directory.
 
     File unEncryptedDir;
     if (briefcaseLfd.isFileEncryptedForm()) {
-      // create or clean-up the temp directory that will hold the unencrypted
+      // create the temp directory that will hold the unencrypted
       // files. Do this in the outputDir so that the briefcase storage location
       // can be a read-only network mount. issue 676.
-      unEncryptedDir = new File(outputDir, ".temp");
-
-      if (unEncryptedDir.exists()) {
-        // silently delete it...
-        try {
-          FileUtils.deleteDirectory(unEncryptedDir);
-        } catch (IOException e) {
-          String msg = "Unable to delete stale temp directory: " + unEncryptedDir.getAbsolutePath();
-          log.warn(msg, e);
-          EventBus.publish(new ExportProgressEvent(msg, briefcaseLfd));
-          return false;
-        }
-      }
-
-      if (!unEncryptedDir.mkdirs()) {
+      unEncryptedDir = null;
+      try {
+        Path path = Files.createTempDirectory(Paths.get(outputDir.getAbsolutePath()), ".temp");
+        unEncryptedDir = path.toFile();
+      } catch (IOException e) {
         EventBus.publish(new ExportProgressEvent("Unable to create temp directory: "
-            + unEncryptedDir.getAbsolutePath(), briefcaseLfd));
+                + unEncryptedDir.getAbsolutePath(), briefcaseLfd));
```

Comment:
Let's improve this event a little bit by adding a cause to the message like in line 828: 
```java
EventBus.publish(new ExportProgressEvent(msg + " Cause: " + e.toString(), briefcaseLfd));
```

Until we refactor error handling, it's the only way we can detect that an `ExportProgressEvent` comes with some negative outcome.

Reply:
:+1: